src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SymbolToProjectId.cs

```diff
@@ -62,31 +63,18 @@ internal partial class SolutionState
                     // Have to lock as there's no atomic AddOrUpdate in netstandard2.0 and we could throw if two
                     // threads tried to add the same item.
 #if NETSTANDARD
-                    lock (assemblyOrModuleSymbolToProjectId)
+                    lock (unrootedSymbolToProjectId)
                     {
-                        assemblyOrModuleSymbolToProjectId.Remove(symbol);
-                        assemblyOrModuleSymbolToProjectId.Add(symbol, projectId);
+                        unrootedSymbolToProjectId.Remove(symbol);
+                        unrootedSymbolToProjectId.Add(symbol, projectId);
                     }
 #else
-                    assemblyOrModuleSymbolToProjectId.AddOrUpdate(symbol, projectId);
+                    unrootedSymbolToProjectId.AddOrUpdate(symbol, projectId);
 #endif
                 }
 
                 return projectId;
             }
-            else if (symbol.IsKind(SymbolKind.DynamicType))
-            {
-                foreach (var (projectId, tracker) in _projectIdToTrackerMap)
-                {
-                    // VB doesn't have DynamicTypes (and throws if you ask for them), so just check C# projects.
-                    if (tracker.TryGetCompilation(out var compilation) &&
```

Comment:
this was hte failure point from before.  it was possible to hold onto a dynamic type, while it's compilation could still get GC'ed underneath it.  so later when we walked the solution looking for it, we wouldn't get a compilatino back from any tracker, and so we wouldn't find this symbol.